File: backend/src/models/notification.ts
import {
  Entity,
  Column,
  PrimaryGeneratedColumn,
  BaseEntity,
  CreateDateColumn,
  UpdateDateColumn
} from 'typeorm';

@Entity()
export class Notification extends BaseEntity {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;

  @Column({
    type: 'timestamp',
    nullable: true
  })
  startDatetime: Date | null;

  @Column({
    type: 'timestamp',
    nullable: true
  })
  endDatetime: Date | null;

  @Column({
    nullable: true,
    type: 'varchar'
  })
  maintenanceType: string | null;

  @Column({
    nullable: true,
    type: 'varchar'
  })
  status: string | null;

  @Column({
    nullable: true,
    type: 'varchar'
  })
  updatedBy: string | null;

  @Column({
    nullable: true,
    type: 'varchar'
  })
  message: string | null;
}

File: backend/src/xfd_django/xfd_api/schema_models/notification.py
"""Notification schema."""
# Standard Python Libraries
from datetime import datetime
from typing import Optional
from uuid import UUID

# Third-Party Libraries
from pydantic import BaseModel


class Notification(BaseModel):
    """Notification schema."""

    id: UUID
    created_at: datetime
    updated_at: datetime
    start_datetime: Optional[datetime]
    end_datetime: Optional[datetime]
    maintenance_type: Optional[str]
    updated_by: Optional[str]
    status: Optional[str]
    message: Optional[str]

    class Config:
        """Config."""

        from_attributes = True


class CreateNotificationSchema(BaseModel):
    """Create notification schema."""

    maintenance_type: str
    status: str
    updated_by: str
    message: str
    start_datetime: datetime
    end_datetime: datetime
